src/librustc_mir/transform/promote_consts.rs

```diff
@@ -280,11 +281,11 @@ impl<'tcx> Validator<'_, 'tcx> {
                             return Err(Unpromotable);
                         }
 
-                        // FIXME(eddyb) compute this on the fly.
-                        let mut has_mut_interior = self.has_mut_interior.contains(base);
+                        let mut has_mut_interior =
```

Comment:
Why not call `validate_local` to check for promotability first instead of at the end of this function? That way you don't have to worry about promotability in `qualif_local` at all, just `Freeze` and `Drop`.

Reply:
Ohh, right, it's so lopsided right now because before the bitsets were cheaper.
I'll flip it around, thanks.